x-pack/plugins/security_solution/public/entity_analytics/components/asset_criticality_selector.tsx

```diff
@@ -0,0 +1,277 @@
+          <EuiLoadingSpinner size="s" />
+        ) : (
+          <EuiFlexGroup direction="row" alignItems="center" wrap={false}>
+            <EuiFlexItem>
+              <EuiText size="s">
+                {criticality.status === 'update' && criticality.query.data?.criticality_level ? (
+                  <EuiHealth
+                    color={CRITICALITY_LEVEL_COLOR[criticality.query.data.criticality_level]}
+                  >
+                    {criticalityDisplayText[criticality.query.data.criticality_level]}
+                  </EuiHealth>
+                ) : (
+                  <EuiHealth color="subdued">{CREATE_ASSET_CRITICALITY}</EuiHealth>
+                )}
+              </EuiText>
+            </EuiFlexItem>
+            <EuiFlexItem>
+              <EuiButtonEmpty
+                data-test-subj="asset-criticality-change-btn"
+                iconType="arrowStart"
+                iconSide="left"
+                flush="right"
+                onClick={criticality.modal.toggle(true)}
+              >
+                {criticality.status === 'update' ? 'Change' : 'Create'}
+              </EuiButtonEmpty>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        )}
+      </EuiAccordion>
+      {criticality.modal.visible ? (
+        <AssetCriticalityModal entity={entity} criticality={criticality} />
+      ) : null}
+    </>
+  );
+};
+
+interface ModalProps {
+  criticality: State;
+  entity: Entity;
+}
+const AssetCriticalityModal: React.FC<ModalProps> = ({ criticality, entity }) => {
+  const basicSelectId = useGeneratedHtmlId({ prefix: 'basicSelect' });
+  const [value, setNewValue] = useState<AssetCriticalityRecord['criticality_level']>(
+    criticality.query.data?.criticality_level ?? 'normal'
+  );
+
+  return (
+    <EuiModal onClose={criticality.modal.toggle(false)}>
+      <EuiModalHeader>
+        <EuiModalHeaderTitle data-test-subj="asset-criticality-modal-title">
+          {PICK_ASSET_CRITICALITY}
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        <EuiSuperSelect
+          id={basicSelectId}
+          options={options}
+          valueOfSelected={value}
+          onChange={setNewValue}
+          aria-label={PICK_ASSET_CRITICALITY}
+          data-test-subj="asset-criticality-modal-select-dropdown"
+        />
+      </EuiModalBody>
+      <EuiModalFooter>
+        <EuiButtonEmpty onClick={criticality.modal.toggle(false)}>{'Cancel'}</EuiButtonEmpty>
+
+        <EuiButton
+          onClick={() =>
+            criticality.mutation.mutate({
+              criticalityLevel: value,
+              idField: `${entity.type}.name`,
+              idValue: entity.name,
+            })
+          }
+          fill
+          data-test-subj="asset-criticality-modal-save-btn"
+        >
+          {'Save'}
+        </EuiButton>
+      </EuiModalFooter>
+    </EuiModal>
+  );
+};
+
+const useAssetCriticality = (entity: Entity): State => {
```

Comment:
Not sure if this is the correct place for this hook?
This is under the `components` folder, maybe we want the hook logic somewhere else?

Reply:
Maybe `security_solution/public/entity_analytics/hooks/use_asset_criticality.ts`?

Reply:
`use_asset_criticality` looks like a single usage hook. You could create a folder for the component for `asset_criticality` inside `components` and encapsulate everything related to the component inside the folder.

Reply:
done.
As discussed I separated the hook into one dealing with data fetching and another for the modal